lib/cocoapods-core/cdn_source.rb

```diff
@@ -30,6 +30,20 @@ def initialize(repo)
       super(repo)
     end
 
+    # @return [Async::HTTP::Internet] The async HTTP client.
+    #
+    def http_client
+      @http_client ||=
+        begin
+          options = {
+            :retries => 0,
```

Comment:
This probably isn't doing what you expect. Are you handling `Protocol::RequestFailed`?

Reply:
We handled `::StandardError` in https://github.com/CocoaPods/Core/pull/669/files/498c56c75481a395731909508a5f57f9db4cdc74#diff-7f437739a49b3cd41100e6671b555ea552bd12a91196f25d2cc7b1d6fd30693cR84

I think It should rescue `Protocol::RequestFailed` too

Reply:
Oh I may misunderstand this, do you mean `Protocol::RequestFailed` raise on `Async::HTTP::Internet.new(**options)` ?

Reply:
Handle `Protocol::RequestFailed` when do the request

Reply:
Persistent connections can go away at any time or break for any reason. It depends on the server. HTTP/1 does not have any good way to inform the client of a graceful disconnect. So, you need to carefully handle this case.